src/core/layertreemodel.cpp

```diff
@@ -102,6 +107,34 @@ QVariant LayerTreeModel::data( const QModelIndex &index, int role ) const
         return node->isVisible();
       }
     }
+
+    case Trackable:
+    {
+      QgsLayerTreeNode *node = mLayerTreeModel->index2node( mapToSource( index ) );
+      if ( QgsLayerTree::isLayer( node ) )
+      {
+        QgsLayerTreeLayer *nodeLayer = QgsLayerTree::toLayer( node );
+        if ( nodeLayer->layer()->type() == QgsMapLayerType::VectorLayer )
+        {
+          QgsVectorLayer *layer = qobject_cast<QgsVectorLayer *>( nodeLayer->layer() );
+          return ( layer->geometryType() == QgsWkbTypes::LineGeometry || layer->geometryType() == QgsWkbTypes::PolygonGeometry );
+        }
+      }
+      return false;
+    }
+
+    case OnTrack:
```

Comment:
```suggestion
    case IsTracking:
```

Reply:
I'm preferring `InTracking` instead of `IsTracking`.
Because the layer is not tracking. A tracking session is tracking on the layer, so I would say like "The layer is currently in a tracking session/state"...